### ISSUE TITLE
Swagger-UI The authorize button is not showing to allow to fill basic auth

### ISSUE THREAD

**Issue by TehBakker** [springfox/springfox#2058]
- Springfox 2.6.1

 - Question . Feature Request
Following #1236 
I've paths that require a basic auth, from my swagger specification : 

`  securityDefinitions:
    basicAuth:
      type: basic
      description: HTTP Basic Authentication.`

   ```
 /configuration:
      get:
        tags:
          - Configuration
        summary: Fetch all configurations
        description: Fetch all the configurations associated with the customer authenticated.
        operationId: getAllConfiguration
        produces:
          - application/json;charset=UTF-8
        security:
          - basicAuth: []
        responses:
          '200':
            description: All the configurations found.
            schema:
              type: array
              items:
                $ref: '#/definitions/CustomerConf'
          '400':
            $ref: '#/responses/IllegalInput'
          '401':
            $ref: '#/responses/NotAuthenticated'
          '404':
            $ref: '#/responses/NotFound'
          '500':
            $ref: '#/responses/InternalError'
```
I see my path fine on swagger ui but when I click the "try it", I get a 401 because I'm not authenticated.
I was expecting that  like default swagger UI, there would be an authorize button or that some fields would show allowing the user to fill the header /  authorization.

I attached 2 capture of what I get with another API not using springfox.

To sum it up, how can I fill basic auth credential via the swagger ui page ?

![swag](https://user-images.githubusercontent.com/7705294/31128036-d85ac3cc-a850-11e7-9283-51b48c4ef8bd.png)
![swag2](https://user-images.githubusercontent.com/7705294/31128041-dae2b91a-a850-11e7-950b-bfb1d9659462.jpg)

**Comment by dilipkrish** [comment]
You need to setup your `SecurityContext` with the right predicate so that it will show up the authorization prompt. It should work like this above.

**Comment by TehBakker** [comment]
Thanks for the answer  dilipkrish.
I'm generating the code from the swagger specification (through swagger codegen).

Does my swagger spec looks like it is missing something or it's a codegen related issue ?

**Comment by dilipkrish** [comment]
It might be a combination of the specification and the codegen.
